aequilibrae/project/network/link.py

```diff
@@ -107,12 +107,12 @@ def add_mode(self, mode: [str, Mode]):
         mode_id = self.__validate(mode)
 
         if mode_id in self.modes:
-            logger.warning("Mode already active for this link")
+            self._logger.warning("Mode already active for this link")
```

Comment:
I don't think that _logger is populated for the link is it? self.project.logger perhaps?